velox/docs/develop/scalar-functions.rst

```diff
@@ -104,6 +104,58 @@ an artificial example of a ceil function that returns 0 for null input:
 Notice that callNullable function takes arguments as raw pointers and not
 references to allow for specifying null values.
 
+Null-Free Fast Path
+*******************
+
+A "callNullFree" function may be implemented in place of or along side "call"
+and/or "callNullable" functions. When only the "callNullFree" function is
+implemented, evaluation of the function will be skipped and null will
+automatically be produced if any of the input arguments are null (like deafult
+null behavior) or if any of the input arguments are of a complex type and
+contain null anywhere in their value, e.g. an array that has a null element.
+If "callNullFree" is implemented alongside "call" and/or "callNullable", an
+efficient check is applied to the batch to see if any of the input arguments
+may be or contain null. Only if it can definitively be determined that that are
+no nulls will "callNullFree" be invoked.  In this case, "callNullFree" can act
+as a fast path by avoiding any per row null checks.
+
+Here is an example of an array_min function that returns the minimum value in
+an array:
+
+.. code-block:: c++
+
+				template <typename TExecParams>
```

Comment:
can this be written with proper indentation.

Reply:
Ya, I'm not sure why some of the other code blocks in this file are indented so far over